lib/matplotlib/tight_layout.py

```diff
@@ -123,6 +123,9 @@ def auto_adjust_subplotpars(fig, renderer,
     for subplots, ax_bbox, (num1, num2) in zip(subplot_list,
                                                ax_bbox_list,
                                                num1num2_list):
+        if all([not ax.get_visible() for ax in subplots]):
```

Comment:
The first commit tried that approach (see [here](https://github.com/matplotlib/matplotlib/pull/8244/commits/3989c06911871e6a890f18cf89aab13ee7e66d91)).

But it actually failed to fix the issue, just down-streamed it. If you try to use that code with the original minimal failing example:

```python
import matplotlib.pyplot as plt
f, axarr = plt.subplots(2, 2)
axarr[1][1].set_visible(False)
plt.tight_layout()
```

The `union` op will fail because the list comprehension will pass it an empty list (`[]`).